src/asl/synth.ts

```diff
@@ -1974,6 +1974,56 @@ export class ASL {
               : ASLGraph.intrinsicStringToJson(output)
           );
         });
+      } else if (isSplit(expr)) {
+        const [splitter, limit] = expr.args;
+
+        if (limit) {
+          throw new SynthError(
+            ErrorCodes.Unsupported_Feature,
+            "Step Functions does not support the limit parameter on String.split"
+          );
+        } else if (!splitter) {
```

Comment:
Can this be supported when it's all literal strings and no references?

Reply:
What is the use case where this would actually matter? When the constant is a reference resolvable at runtime? I'll try it.